### PR TITLE
misc(audits): removed unused audit meta.category

### LINKED ISSUE #3252: Remove audit.meta.category
As discussed here https://github.com/GoogleChrome/lighthouse/pull/3227#discussion_r137057437 .. it's not needed and we can drop it.

### PR DESCRIPTION
**Pull request by arturmiz**
Closes #3252

### CODE REVIEW

**Comment by paulirish** [comment]
Hero!

**Comment by patrickhulce** [comment]
amazing! 👏

File: lighthouse-cli/test/smokehouse/smokehouse.js [hunk 1]
```diff
@@ -196,7 +196,6 @@ function collateResults(actual, expected) {
     const diff = findDifference(auditName, actualResult, expectedResult);
 
     return {
-      category: auditName,
```
Comment (brendankenny) [inline lighthouse-cli/test/smokehouse/smokehouse.js]: the two in this file are a different `category` and should stay :)

File: lighthouse-core/test/config/config-test.js [hunk 2]
```diff
@@ -212,10 +211,6 @@ describe('Config', () => {
     }), /audit\(\) method/);
 
     assert.throws(_ => new Config({
-      audits: [basePath + '/missing-category'],
```
Comment (brendankenny) [inline lighthouse-core/test/config/config-test.js]: will you also delete the file this was loading (`lighthouse-core/test/fixtures/invalid-audits/missing-category.js`)

**Comment by brendankenny** [comment]
one more in `https://github.com/GoogleChrome/lighthouse/blob/487bee6652fb3780f54bd05c4767fbd72eaca7cf/lighthouse-cli/types/types.ts` (though not sure if that file is maintained :)

**Comment by arturmiz** [comment]
@brendankenny code review fixes applied 👍

**Pull request merged**